src/particle_exchanger.h

```diff
@@ -154,28 +155,22 @@ void ParticleExchanger_t<Halo_T>::CollectParticles()
 
 template <class Halo_T>
 void ParticleExchanger_t<Halo_T>::SendParticles()
-{ // order the particles and send them to the corresponding processing according to ProcessIdRange
-  sort(LocalParticles.begin(), LocalParticles.end(), ParticleExchangeComp::CompParticleId);
+{
+
+  // Sort the particles by destination rank
+  std::vector<HBTInt> offset = sort_by_hash(LocalParticles, world.size());
 
   LocalSizes.resize(world.size());
   LocalIterators.resize(world.size());
-  int rank = 0;
-  for (auto it = LocalParticles.begin(); it != LocalParticles.end(); ++it)
+  int rank;
```

Comment:
Might be better to limit the definition of `rank` within the scope of the relevant block of code, e.g.
```
```suggestion
  int rank;
```
{
    int rank;
    for (rank = 0; rank < world.size(); rank++)
    {
        LocalIterators[rank] = LocalParticles.begin() + offset[rank];
    }
    for (rank = 0; rank < world.size() - 1; rank++)
    {
        LocalSizes[rank] = LocalIterators[rank + 1] - LocalIterators[rank];
    }
    LocalSizes[rank] = LocalParticles.end() - LocalIterators.back();
}
```

Reply:
Ok. Need to be a little bit careful here because the code was relying on the left over value of rank after the loop.